src/cmd-cloud-prune

```diff
@@ -414,5 +437,71 @@ def prune_build(s3_client, bucket, prefix, build_id, dry_run):
                 raise Exception(f"Error pruning {build_id}: {e.response['Error']['Message']}")
 
 
+def get_container_tags(meta_json, stream):
+    container_tags = []
+    base_oscontainer = meta_json.get("base-oscontainer")
+    if base_oscontainer:
+        tags = base_oscontainer.get("tags", [])
+        # Only include tags that do not match the stream i.e. moving tags
+        filtered_tags = [tag for tag in tags if tag != stream]
+        if filtered_tags:
+            container_tags = filtered_tags
+    return container_tags
+
+
+def prune_containers(containers_config):
+    barrier_releases = set()
+    # Get the update graph for stable streams
+    if containers_config["stream"] in ['stable', 'testing', 'next']:
+        update_graph = get_update_graph(containers_config["stream"])['releases']
+        # Keep only the barrier releases
+        barrier_releases = set([release["version"] for release in update_graph if "barrier" in release])
+
+    for tag in containers_config["container_tags"]:
+        if tag in STREAMS:
+            continue
+        if tag in barrier_releases:
+            print(f"Release {tag} is a barrier release, keeping.")
+            continue
```

Comment:
This assumes that the tag is the version number, which happens to be true, but I'd rather it not rely on that assumption. As mentioned in my comment above let's just skip container pruning if the build we are on is a barrier release (i.e. way before we ever get into this code).

Reply:
I am not sure I get "not assuming the tag as a version number".
I am trying to do this check way above but the check itself would still be similar right?
```
if tag in barrier_releases:
    print(f"Release {tag} is a barrier release, keeping.")
else:
    prune_container(tag, containers_config)
```

Reply:
just comparing it to see if the build_id is in the list of barrier releases should work.

Reply:
resolved in new upload